pkg/agent/runner.go

```diff
@@ -194,6 +196,9 @@ func (r *Runner) Run(ctx context.Context, logger *zap.Logger, vmInfoUpdated util
 	pluginRequestJitter := util.NewTimeRange(time.Millisecond, 0, 100).Random()
 
 	coreExecLogger := execLogger.Named("core")
+	revisionSource := revsource.NewRevisionSource(
+		WrapHistogramVec(&r.global.metrics.scalingLatency),
+	)
```

Comment:
Can we pass in the VM spec's current revision here so that we pick up where the previous agent left off? (otherwise IIUC we'll get a bunch of warnings on restart, right?)

Reply:
That's a great idea! Implemented it in 0df27ac9974d4e06dd32eb186053e1e062c3d00f, take a look, pls.

Reply:
nice! One nit, left a separate comment about it